### DISCUSSION TITLE
Should Speak Touch announce the hint for elements other than the edit field?

### DISCUSSION THREAD

**Discussion by Irineu333** [Irineu333/SpeakTouch#121] [Question]
Starting from pull request #108, Speak Touch announces the hintText only for elements of the edit field type.

**Comment by PatrykMis** [comment]
I'm not sure. Buttons can also have hint text. For example, the WhatsApp "voice message" button which text hint says something like "double tap and hold to record, pull to send, move down to dismiss ...". But I'm not sure if it's related specifically to `hint` attribute.

**Comment by PatrykMis** [comment]
Similar behavior have some elements in facebook app. There was something like "double tap and hold two fingers to generate image description" on an image element. The hint quotes may be definitely not accurate, but I am trying to explain how it may work.

**Comment by PatrykMis** [comment]
From what I've tested, currently SpeakTouch from #108 reads properly descriptions of the buttons like mentioned from WhatsApp and Facebook; so whether hints for other controlls are being read or not - it works like it should.

**Reply by Irineu333** [reply]
Great! If it's being announced, it's because it's in the `contentDescription`, not in the `hintText`. From what I understand, `hintText` is specific for edit fields. For other elements, whose tip is related to another type of interaction, there are the usage tips, which is nothing more than announcing the actions for that node. For example, you can customize the `ACTION_LONG_CLICK` action, and instead of the app announcing "double-tap and hold to press", you can make it announce "double-tap and hold to select", which is a more precise description [^1]. Speak Touch still does not support actions or usage tips, but we have already mapped this functionality.

[^1]: https://developer.android.com/guide/topics/ui/accessibility/principles#understandable-actions
